### ISSUE TITLE
Bug: useDark() is not responsive with multiple instances

### ISSUE THREAD

**Issue by shigma** [vueuse/vueuse#2930]
### Describe the bug

I used `useDark()` across different components and found that `useDark()` is not responsive to each other.

### Reproduction

https://stackblitz.com/edit/vitejs-vite-abnadb?file=src/App.vue

### System Info

```Shell
System:
    OS: macOS 12.6.3
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 75.80 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.14.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.3.1 - /usr/local/bin/npm
  Browsers:
    Chrome: 111.0.5563.146
    Safari: 16.3
```

### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/vueuse/vueuse/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a VueUse issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/core instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vueuse/vueuse/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by Waleed-KH** [comment]
`useDark()` return a new instance each time you call it, so if you need to use it in different places or files, you need to separate the code in a different file and export the instance, then import it where ever you need, example: https://stackblitz.com/edit/vitejs-vite-gx9vzw?file=src%2FApp.vue,src%2Ftheme.js

**Comment by shigma** [comment]
> `useDark()` return a new instance each time you call it, so if you need to use it in different places or files, you need to separate the code in a different file and export the instance, then import it where ever you need, example: [stackblitz.com/edit/vitejs-vite-gx9vzw?file=src%2FApp.vue,src%2Ftheme.js](https://stackblitz.com/edit/vitejs-vite-gx9vzw?file=src%2FApp.vue,src%2Ftheme.js)

I know, but this does not make a lot of sense to me. `useDark()` use `useStorage()` and `useEventListener()` under the hood, which means it could be and should be responsive to each other. If one instance makes some changes to the localStorage or root element, other instances looks quite weird.

**Comment by Waleed-KH** [comment]
Yes, they use the same key and they should be in sync, but you are using an old version (v7.6.2), there was a bug in `useStorage()` [which was fixed in v9.8.2](https://github.com/vueuse/vueuse/commit/800f74f49197b11ee5691ac8f4061eb2459ad07e), just update the packages and your code will work: https://stackblitz.com/edit/vitejs-vite-f8atgi?file=package.json,src%2FApp.vue
even though, I would still recommend the approach above

**Comment by shigma** [comment]
I see. Thanks for your response.
